src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.epics.js

```diff
@@ -11,56 +12,97 @@ import {
     showLoadingViewOnEnrollmentPage,
     successfulFetchingEnrollmentPageInformationFromUrl,
     openEnrollmentPage,
+    startFetchingTeiFromEnrollmentId,
+    startFetchingTeiFromTeiId,
 } from './EnrollmentPage.actions';
 import { urlArguments } from '../../../utils/url';
 
 const fetchEnrollment = id => getApi().get(`enrollments/${id}`, { fields: 'trackedEntityInstance,program,orgUnit' });
 const fetchTrackedEntityInstance = id => getApi().get(`trackedEntityInstances/${id}`, { fields: 'attributes,enrollments' });
 
+const sortByDate = (enrollments = []) => enrollments.sort((a, b) =>
+    moment.utc(b.enrollmentDate).diff(moment.utc(a.enrollmentDate)));
+const deriveSelectedName = (attributes = {}) => attributes.reduce((acc, { value: dataElementValue }) =>
+    (acc ? `${acc} ${dataElementValue}` : dataElementValue), '');
+
+
+const fetchTeiStream = teiId => from(fetchTrackedEntityInstance(teiId))
+    .pipe(
+        map(({ attributes, enrollments }) => {
+            // todo this is not scaling when you have many attributes the name will be huge
```

Comment:
Still need to filter here for the first two of the list as we discused so that we display some meaningful name. 
I will tackle this together with this subtask https://jira.dhis2.org/browse/DHIS2-10370